src/effects.cpp

```diff
@@ -176,55 +176,162 @@ void LoadEffectFactories()
 
     // #define EFFECT_SET_VERSION   0
 
+    // Include custom effects header if available - it overrides whatever the effect set flags
+    // would otherwise include.
+    //
+    // REVIEW (davepl) This allows the user to define custom effects
+    // in a separate file, which is included here. This way, the user can add custom effects
+    // without having to modify the main effect factories code, which is useful for projects that
+    // are shared across multiple boards and/or projects.  But it replaces ALL effects.  If we
+    // want to allow custom effects to be added to the existing set, we need to change this
+    // not to have an else case after but rather an endif.
+
+    // REVIEW (davepl) If someone were to define multiple effects sets, it would overwrite the
+    // value of EFFECT_SET_VERSION differently for each set, which is not what we want.
+    //
+
+
     #if __has_include ("custom_effects.h")
 
       #include "custom_effects.h"
 
-    // Fill effect factories
-    #elif M5DEMO
+    // Fill effect factories using new effect set flags
+
+    // === EFFECT SETS ===
+    // These sections are shared by multiple projects
+
+    #elif defined(EFFECTS_MINIMAL)

```

Comment:
Just mentioning here that if any of the "effective effects sets" for any of the known project names (`-DPROJECT_NAME` in platformio.ini) has changed in this update, the `EFFECT_SET_VERSION` define for that block needs to be bumped, to make the people flashing this update get the updated set. If `PROJECT_NAME` and `EFFECT_SET_VERSION` are the same as what had been persisted to JSON with the then-current effect set, the device will load the set persisted to JSON instead of what is put together here.